saq/queue/postgres.py

```diff
@@ -81,10 +76,29 @@ class PostgresQueue(Queue):
     """
 
     @classmethod
-    def from_url(cls: type[PostgresQueue], url: str, **kwargs: t.Any) -> PostgresQueue:
-        """Create a queue from a postgres url."""
+    def from_url(
+        cls: type[PostgresQueue], url: str, min_size: int = 4, max_size: int = 20, **kwargs: t.Any
+    ) -> PostgresQueue:
+        """Create a queue from a postgres url.
+
+        Args:
+            url: The postgres url.
+            min_size: minimum pool size. (default 4)
+                The minimum number of Postgres connections.
+            max_size: maximum pool size. (default 20)
+            If greater than 0, this limits the maximum number of connections to Postgres.
+            Otherwise, maintain `min_size` number of connections.
+
+        kwargs: Additional keyword arguments to pass to the constructor.
+        """
         return cls(
-            AsyncConnectionPool(url, check=AsyncConnectionPool.check_connection, open=False),
+            AsyncConnectionPool(
+                url,
+                min_size=min_size,
+                max_size=max_size,
+                check=AsyncConnectionPool.check_connection,
+                open=True,
```

Comment:
why did you change this open=True?

Reply:
I have removed how I do this, will need a completely new look